Yesod/Paginator/Widget.hs

```diff
@@ -115,12 +117,20 @@ paginationWidget (PageWidgetConfig {..}) page per tot = do
 
                 -- the middle lists, strip the first/last pages and
                 -- correctly take up to limit away from current
-                prevLinks = reverse . take pageCount . reverse . drop 1 $ map (\p -> Enabled p (showT p) "prev") prev
-                nextLinks = take pageCount . reverse . drop 1 . reverse $ map (\p -> Enabled p (showT p) "next") next
+                prevLinks = reverse . take pageCount . reverse . drop 1 $ map (display "prev") prev
+                nextLinks = take pageCount . reverse . drop 1 . reverse $ map (display "next") next
 
                 -- finally, this page itself
                 curLink = [Disabled (showT pg) "active"]
 
+                pageWithinBounds p = case maxPageLinks of
+                    Nothing -> True
+                    Just m  -> if p > m then False
+                                       else True
+                display cls p = if (pageWithinBounds p)
```

Comment:
I think if you reverse these arguments to `display p cls` you'll find `display` is now basically a drop-in for `Enabled`. I think that'll make for a clearer diff that is mostly (if not entirely) just replacing uses of `Enabled` with `display` throughout this function. At that point, maybe a better name will spring to mind.